src/porepy/models/run_convergence_study.py

```diff
@@ -0,0 +1,217 @@
+logger = logging.getLogger(__name__)
+
+
+def gb_refinements(
+        network: Union[pp.FractureNetwork3d, pp.FractureNetwork2d],
+        gmsh_folder_path: Union[str, Path],
+        mesh_args: dict,
+        n_refinements: int = 0
+) -> List[pp.GridBucket]:
+    """ Create n refinements of a fracture network.
+
+    Parameters
+    ----------
+    network : Union[pp.FractureNetwork3d, pp.FractureNetwork2d]
+        Fracture network
+    gmsh_folder_path : str or pathlib.Path
+        Absolute path to folder to store results in
+    mesh_args : dict
+        Arguments for meshing (of coarsest grid)
+    n_refinements : int, Default = 0
+        Number of refined grids to produce.
+        The grid is refined by splitting.
+    """
+
+    # ----------------------------------------
+    # --- CREATE FRACTURE NETWORK AND MESH ---
+    # ----------------------------------------
+    if isinstance(network, pp.FractureNetwork2d):
+        dim = 2
+    elif isinstance(network, pp.FractureNetwork3d):
+        dim = 3
+    else:
+        # This might be too strict: consider passing dim as parameter instead.
+        raise ValueError("Unknown input network")
+
+
+    gmsh_file_name = str(gmsh_folder_path / "gmsh_frac_file")
+    in_file = f'{gmsh_file_name}.geo'
+    out_file = f"{gmsh_file_name}.msh"
+
+    # We need to generate a .geo file for refine_mesh. But that method will mesh the coarsest grid for us,
+    # so we don't need to call network.mesh(...). Instead, we emulate its behaviour except for meshing itself.
+    # -- Start of FractureNetwork3d.mesh(...)
+    if not network.bounding_box_imposed:
+        network.impose_external_boundary(network.domain)
+
+    # Find intersections between fractures
+    if not network.has_checked_intersections:
+        network.find_intersections()
+    else:
+        logger.info("Use existing intersections")
+
+    if "mesh_size_frac" not in mesh_args.keys():
+        raise ValueError("Meshing algorithm needs argument mesh_size_frac")
+    if "mesh_size_min" not in mesh_args.keys():
+        raise ValueError("Meshing algorithm needs argument mesh_size_min")
+
+    mesh_size_frac = mesh_args.get("mesh_size_frac", None)
+    mesh_size_min = mesh_args.get("mesh_size_min", None)
+    mesh_size_bound = mesh_args.get("mesh_size_bound", None)
+    network._insert_auxiliary_points(mesh_size_frac, mesh_size_min, mesh_size_bound)
+
+    # Process intersections to get a description of the geometry in non-intersecting lines and polygons
+    network.split_intersections()
+
+    # Dump the network description to gmsh .geo format
+    network.to_gmsh(in_file, in_3d=True)
+    # -- End of method: .geo file created
+
+    gb_list = refine_mesh(
+        in_file=in_file,
+        out_file=out_file,
+        dim=dim,
+        network=network,
+        num_refinements=n_refinements,
+    )
+    for _gb in gb_list:
+        pp.contact_conditions.set_projections(_gb)
+
+    return gb_list
+
+
+def run_model_for_convergence_study(
```

Comment:
are you computing the L2 errors? if so it should be clear, otherwise is a bit misleading

Reply:
I agree.

I also think we can generalize the error computations if need be, but that is not for this PR, nor the task of @haakon-e